scaladoc/src/dotty/tools/scaladoc/snippets/SnippetCompiler.scala

```diff
@@ -49,17 +50,20 @@ class SnippetCompiler(
   private def nullableMessage(msgOrNull: String): String =
     if (msgOrNull == null) "" else msgOrNull
 
-  private def createReportMessage(wrappedSnippet: WrappedSnippet, arg: SnippetCompilerArg, diagnostics: Seq[Diagnostic]): Seq[SnippetCompilerMessage] = {
+  private def createReportMessage(wrappedSnippet: WrappedSnippet, arg: SnippetCompilerArg, diagnostics: Seq[Diagnostic], sourceFile: SourceFile): Seq[SnippetCompilerMessage] = {
     val line = wrappedSnippet.outerLineOffset
     val column = wrappedSnippet.outerColumnOffset
     val innerLineOffset = wrappedSnippet.innerLineOffset
     val innerColumnOffset = wrappedSnippet.innerColumnOffset
     val infos = diagnostics.toSeq.sortBy(_.pos.source.path)
     val errorMessages = infos.map {
       case diagnostic if diagnostic.position.isPresent =>
-        val diagPos = diagnostic.position.get
+        val diagPos = diagnostic.position.get.asInstanceOf[dotty.tools.dotc.util.SourcePosition]
```

Comment:
Could we narrow the type of the `position` in the case pattern instead of doing a cast?

Reply:
Changed